File: apiwdbe/WdbeQPrcKHdltype.cpp
/**
	* \file WdbeQPrcKHdltype.cpp
	* API code for table TblWdbeQPrcKHdltype (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQPrcKHdltype.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQPrcKHdltype
 ******************************************************************************/

WdbeQPrcKHdltype::WdbeQPrcKHdltype(
			const uint jnum
			, const string yesnoFixed
			, const string sref
			, const string Avail
			, const string Implied
			, const string Title
			, const string Comment
		) :
			jnum(jnum)
			, yesnoFixed(yesnoFixed)
			, sref(sref)
			, Avail(Avail)
			, Implied(Implied)
			, Title(Title)
			, Comment(Comment)
		{
};

bool WdbeQPrcKHdltype::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQPrcKHdltype");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "yesnoFixed", "fix", yesnoFixed);
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Avail", "avl", Avail);
		extractStringUclc(docctx, basexpath, "Implied", "imp", Implied);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "Comment", "cmt", Comment);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQPrcKHdltype
 ******************************************************************************/

ListWdbeQPrcKHdltype::ListWdbeQPrcKHdltype() {
};

ListWdbeQPrcKHdltype::ListWdbeQPrcKHdltype(
			const ListWdbeQPrcKHdltype& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQPrcKHdltype(*(src.nodes[i]));
};

ListWdbeQPrcKHdltype& ListWdbeQPrcKHdltype::operator=(
			const ListWdbeQPrcKHdltype& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQPrcKHdltype(*(src.nodes[i]));

	return *this;
};

ListWdbeQPrcKHdltype::~ListWdbeQPrcKHdltype() {
	clear();
};

void ListWdbeQPrcKHdltype::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQPrcKHdltype::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQPrcKHdltype* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQPrcKHdltype");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQPrcKHdltype", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQPrcKHdltype(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQPrcKHdltype[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwdbe/WdbeQPrcKHdltype.java
/**
	* \file WdbeQPrcKHdltype.h
	* Java API code for record of table TblWdbeQPrcKHdltype
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQPrcKHdltype {
	
	public WdbeQPrcKHdltype(
				int jnum
				, String yesnoFixed
				, String sref
				, String Avail
				, String Implied
				, String Title
				, String Comment
			) {
		this.jnum = jnum;
		this.yesnoFixed = yesnoFixed;
		this.sref = sref;
		this.Avail = Avail;
		this.Implied = Implied;
		this.Title = Title;
		this.Comment = Comment;
	};

	public int jnum;
	public String yesnoFixed;
	public String sref;
	public String Avail;
	public String Implied;
	public String Title;
	public String Comment;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQPrcKHdltype");

		if (Xmlio.checkXPath(doc, basexpath)) {
			yesnoFixed = Xmlio.extractStringUclc(doc, basexpath, "yesnoFixed", "fix", null, 0);
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Avail = Xmlio.extractStringUclc(doc, basexpath, "Avail", "avl", null, 0);
			Implied = Xmlio.extractStringUclc(doc, basexpath, "Implied", "imp", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			Comment = Xmlio.extractStringUclc(doc, basexpath, "Comment", "cmt", null, 0);
			
			return true;
		};

		return false;
	};
};
